src/components/utils/ManagedTree.js

```diff
@@ -10,10 +10,29 @@ let ManagedTree = React.createClass({
   getInitialState() {
     return {
       expanded: new Set(),
-      focusedItem: null
+      focusedItem: null,
+      listItems: []
     };
   },
 
+  componentWillReceiveProps(nextProps) {
+    this.setState({ listItems: nextProps.listItems });
+  },
+
+  componentDidUpdate(nextProps) {
+    const listItemsLocal = this.state.listItems;
+    if (listItemsLocal && listItemsLocal.length > 0) {
+      if (listItemsLocal.length == 1) {
+        this.focusItem(listItemsLocal[0]);
+      }
+      else {
+        this.expandListItems(listItemsLocal);
+      }
+      listItemsLocal.splice(listItemsLocal.length - 1, 1);
+      this.setState({ listItems: listItemsLocal });
```

Comment:
this seems weird to me... perhaps we do need some state, but the name collision makes it hard to follow...

Also, if we have an expanded list, do we need this listItemsLocal list... and if it does to an empty list because it keeps getting spliced down, can we do this another way. I guess i'm saying, this seems weird :)